src/main/java/com/google/firebase/remoteconfig/FirebaseRemoteConfig.java

```diff
@@ -0,0 +1,121 @@
+
+  @VisibleForTesting
+  FirebaseRemoteConfig(FirebaseApp app, FirebaseRemoteConfigClient client) {
+    this.app = checkNotNull(app);
+    this.remoteConfigClient = checkNotNull(client);
+  }
+
+  private FirebaseRemoteConfig(FirebaseApp app) {
+    this(app, FirebaseRemoteConfigClientImpl.fromApp(app));
+  }
+
+  /**
+   * Gets the {@link FirebaseRemoteConfig} instance for the default {@link FirebaseApp}.
+   *
+   * @return The {@link FirebaseRemoteConfig} instance for the default {@link FirebaseApp}.
+   */
+  public static FirebaseRemoteConfig getInstance() {
+    return getInstance(FirebaseApp.getInstance());
+  }
+
+  /**
+   * Gets the {@link FirebaseRemoteConfig} instance for the specified {@link FirebaseApp}.
+   *
+   * @return The {@link FirebaseRemoteConfig} instance for the specified {@link FirebaseApp}.
+   */
+  public static synchronized FirebaseRemoteConfig getInstance(FirebaseApp app) {
+    FirebaseRemoteConfigService service = ImplFirebaseTrampolines.getService(app, SERVICE_ID,
+            FirebaseRemoteConfigService.class);
+    if (service == null) {
+      service = ImplFirebaseTrampolines.addService(app, new FirebaseRemoteConfigService(app));
+    }
+    return service.getInstance();
+  }
+
+  /**
+   * Gets the current active version of the Remote Config template.
+   *
+   * @return A {@link RemoteConfigTemplate}.
+   * @throws FirebaseRemoteConfigException If an error occurs while getting the template.
+   */
+  public RemoteConfigTemplate getTemplate() throws FirebaseRemoteConfigException {
+    return getTemplateOp().call();
+  }
+
+  /**
+   * Similar to {@link #getTemplate()} but performs the operation asynchronously.
+   *
+   * @return An {@code ApiFuture} that will complete with a {@link RemoteConfigTemplate} when
+   *      the template is available.
+   */
+  public ApiFuture<RemoteConfigTemplate> getTemplateAsync() {
+    return getTemplateOp().callAsync(app);
+  }
+
+  private CallableOperation<RemoteConfigTemplate, FirebaseRemoteConfigException> getTemplateOp() {
+    final FirebaseRemoteConfigClient remoteConfigClient = getRemoteConfigClient();
+    return new CallableOperation<RemoteConfigTemplate, FirebaseRemoteConfigException>() {
+      @Override
+      protected RemoteConfigTemplate execute() throws FirebaseRemoteConfigException {
+        return remoteConfigClient.getTemplate();
+      }
+    };
+  }
+
+  @VisibleForTesting
+  FirebaseRemoteConfigClient getRemoteConfigClient() {
+    return remoteConfigClient;
+  }
+
+  private static class FirebaseRemoteConfigService extends FirebaseService<FirebaseRemoteConfig> {
+
+    FirebaseRemoteConfigService(FirebaseApp app) {
+      super(SERVICE_ID, new FirebaseRemoteConfig(app));
+    }
+
+    @Override
+    public void destroy() {
+      // NOTE: We don't explicitly tear down anything here, but public methods of
+      // FirebaseRemoteConfig will now fail because calls to getOptions() and getToken()
```

Comment:
What does "now" suggest -- the state if destroy() is called?

I'm trying to figure out if we need all the "will" future tense formations :)

Reply:
I don't think this is publicly visible in docs. It is a comment in code. So we should be good here :)